### PR TITLE
Add SequenceExample viewing support to What-If Tool

### PR DESCRIPTION
**Pull request by jameswex**
- Added checkbox in WIT loader to specify if examples are sequence examples
- Added URL params for checkboxes in WIT loader, matching other settings
- Correctly load and update SequenceExamples in python code
- Correctly parse and display SequenceExamples in javascript code
- Also updated inference display logic to not fail when ground truth label feature is missing

With this change, WIT can display sequence examples, but it still cannot use them for inference in models, as the Classification and Regression APIs do not support them. Will unlock their use in models in future PRs.

### CODE REVIEW

**Comment by jameswex** [comment]
For #1502

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-inference-panel.html [hunk 1]
```diff
@@ -73,7 +73,7 @@
       .model-type-label {
         padding-top: 10px;
       }
-      .multi-class-checkbox {
+      .checkbox {
```
Comment (stephanwlee) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-inference-panel.html]: nit: you can just use `paper-checkbox` probably

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 2]
```diff
@@ -3126,7 +3132,7 @@ <h2>Create a distance feature</h2>
 
       makeAsyncRequest_: function(url, thenDoFn, postData){
         var self = this;
```
Comment (stephanwlee) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: Using arrow function, now, you don't need this `self`

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 3]
```diff
@@ -3138,17 +3144,65 @@ <h2>Create a distance feature</h2>
             thenDoFn(result);
           }
         });
-        this.requestManager.request(url, postData).then(wrapperFn);
+        this.requestManager.request(url, postData).then(wrapperFn)
+            .catch(reason => {
+              this.exampleStatusStr = 'Request failed'
```
Comment (stephanwlee) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: nit: add semi-colon at the end for consistency

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 4]
```diff
@@ -3138,17 +3144,65 @@ <h2>Create a distance feature</h2>
...
+          });
+        }
+        if (v) {
+          // For value lists that are of length 1, treat the feature value
+          // as just the item and not the list of length 1.
+          if (squeeze && v.length == 1) {
+            v = v[0];
+            // For numeric feature values, convert the string to a number.
+            if (!isNaN(v)) {
+              v = Number(v);
```
Comment (stephanwlee) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]:
```
> isNaN('a')
< true
> Number('a')
< NaN
```

I believe it is more kosher to use `parseInt(str, 10)` to cast string to number.

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 5]
```diff
@@ -3138,17 +3144,65 @@ <h2>Create a distance feature</h2>
...
+            }
+            try {
+              return atob(val);
+            } catch (err) {
+              console.error('Error decoding string: ' + err);
+              return 'ERROR: error decoding string';
+            }
+          });
+        }
+        if (v) {
```
Comment (stephanwlee) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: For easier readability, can you use a separate and descriptive variable name for `v`? Reading between lines, it seems like `v` up to this line is regarded as string (or null) and we are returning oneOf `[null, NaN, number, string]` (string in case squeeze == false)

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 6]
```diff
@@ -3138,17 +3144,65 @@ <h2>Create a distance feature</h2>
...
-       * Dive.
+       * Converts a single TF Example or SequenceExample into an item dict for
+       * display in Facets Dive.
        */
       exampleToDataPoint_: function(example, index) {
         const d = {};
+        // Extract the standard features from examples or context features from
+        // sequence examples.
+        const featureRoot = example.features || example.context;
         const features = 'features' in example ?
```
Comment (stephanwlee) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: You may need to update this condition

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-inference-panel.html [hunk 7]
```diff
@@ -105,9 +105,14 @@
...
     </paper-radio-group>
-    <paper-input always-float-label label="Path to examples"
-                 value="{{examplesPath}}">
-    </paper-input>
+    <div class="flex-holder">
+      <paper-input always-float-label label="Path to examples"
+                  value="{{examplesPath}}" class="flex-grow">
+      </paper-input>
+      <paper-checkbox checked="{{sequenceExamples}}" class="checkbox">
+        SequenceExamples
```
Comment (stephanwlee) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-inference-panel.html]: would you want a whitespace between two words instead of CamelCase?

File: tensorboard/plugins/interactive_inference/interactive_inference_plugin.py [hunk 8]
```diff
@@ -55,6 +55,7 @@ class InteractiveInferencePlugin(base_plugin.TBPlugin):
   examples = []
   updated_example_indices = set()
   sprite = None
+  example_class = tf.train.Example
```
Comment (stephanwlee) [inline tensorboard/plugins/interactive_inference/interactive_inference_plugin.py]: Unsure but first character of the class name may need to be capitalized instead of snake_case

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html (hunk 2 above)
Comment (jameswex) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: done

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html (hunk 3 above)
Comment (jameswex) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: done

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html (hunk 5 above)
Comment (jameswex) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: clarified with vars valueList and singleValue

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html (hunk 4 above)
Comment (jameswex) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: I don't want to just cast ints, must handle floats too

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-inference-panel.html (hunk 7 above)
Comment (jameswex) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-inference-panel.html]: I used camel case since SequenceExample is the proto message name that this checkbox is for.

File: tensorboard/plugins/interactive_inference/interactive_inference_plugin.py (hunk 8 above)
Comment (jameswex) [reply tensorboard/plugins/interactive_inference/interactive_inference_plugin.py]: true, changed to adhere to style guide

Comment (jameswex) [reply tensorboard/plugins/interactive_inference/interactive_inference_plugin.py]: Seems like a conflict between style for member variables (snake) and class names (CamelCase), of which this is both. Will keep as this to make it clear its a member variable, since the name documents that it is a class, if that is ok

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html (hunk 6 above)
Comment (jameswex) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: good catch
